api/desecapi/views.py

```diff
@@ -247,34 +248,26 @@ def get_queryset(self):
 
     def get_object(self):
         queryset = self.filter_queryset(self.get_queryset())
-
-        rrset_type = self.kwargs['type']
-
-        if rrset_type in RRset.RESTRICTED_TYPES:
-            raise PermissionDenied("You cannot tinker with the %s RRset." % rrset_type)
-
-        obj = get_object_or_404(queryset, type=rrset_type, subname=self.kwargs['subname'])
-
+        obj = get_object_or_404(queryset, type=self.kwargs['type'], subname=self.kwargs['subname'])
         self.check_object_permissions(self.request, obj)
-
         return obj
 
     def update(self, request, *args, **kwargs):
-        if not isinstance(request.data, dict):
-            raise ValidationError({
-                api_settings.NON_FIELD_ERRORS_KEY: ['Invalid input, expected a JSON object.']
-            }, code='invalid')
-
         # Attach URL parameters (self.kwargs) to the request body object (request.data),
         # the latter having preference with both are given.
+        data = deepcopy(request.data)
         for k in ('type', 'subname'):
-            # This works because we exclusively use JSONParser which causes request.data to be
-            # a dict (and not an immutable QueryDict, as is the case for other parsers)
-            request.data[k] = request.data.pop(k, self.kwargs[k])
+            data[k] = request.data.pop(k, self.kwargs[k])
+
+        partial = kwargs.pop('partial', False)
+        instance = self.get_object()
+        serializer = self.get_serializer(instance, data=data, partial=partial)
```

Comment:
This used to be called indirectly via `super().update(request, *args, **kwargs)` (see a few lines down). We now discard any extra `kwargs`. Should we check here whether any where present, in case something changes in DRF?

Reply:
I believe with regard to the kwargs, the behavior is same as the DRF implementation:

	    def update(self, request, *args, **kwargs):
		partial = kwargs.pop('partial', False)
		instance = self.get_object()
		serializer = self.get_serializer(instance, data=request.data, partial=partial)
		serializer.is_valid(raise_exception=True)
		self.perform_update(serializer)

		if getattr(instance, '_prefetched_objects_cache', None):
		    # If 'prefetch_related' has been applied to a queryset, we need to
		    # forcibly invalidate the prefetch cache on the instance.
		    instance._prefetched_objects_cache = {}

		return Response(serializer.data)

I changed the argument to `get_serializer` to avoid mutating the `request` object, hence so much copy and paste (which I feel bad for). However, I think not changing the `request` object is a good thing.